tests/cpp_compat_tests.rs

```diff
@@ -0,0 +1,48 @@
+#![cfg(feature = "rust")]
+
+use fastpfor::cpp::{Codec32, FastPFor128Codec};
+use fastpfor::rust::{FastPFOR, Integer, BLOCK_SIZE_128, DEFAULT_PAGE_SIZE};
```

Comment:
you are dealing with two separate implementations, so you don't want to accidentally confuse which identifier is from where.  With this change, all relevant idents will have to be language-prefixed, e.g. `cpp::Codec32` and `rust::FastPFOR` (you will need to update the rest of the code)

```suggestion
use fastpfor::{cpp, rust};
```

Reply:
The traits `Codec32` and `Integer` are still needed. Yes, I agree that specifying the model is better.